### ISSUE TITLE
Convert exportImage.js to Reason

### ISSUE THREAD

**Issue by dylanirlbeck** [draftbit/avatar-generator#28]
This file is still in JS because it (a) uses `async/await` syntax and (b) uses a lot of DOM-related methods. We'll need to create bindings for the `html2canvas` package as well, but that shouldn't be too difficult.

Once the rewrite in Reason is done, we can remove the file and keep all the functionality for export inside `src/pages/Index.re`.

**Comment by diogomqbm** [comment]
Is it ok to use [bs-let](https://github.com/reasonml-labs/bs-let) and [bs-wepapi](https://github.com/reasonml-community/bs-webapi-incubator/tree/master/src/Webapi/Webapi__Dom) to work with `async/await` and with all DOM-related methods?

**Comment by dylanirlbeck** [comment]
@diogomqbm Of course! You'll definitely need to use `bs-webapi` (I mean, you _could_ get away without it, but please use it!); in terms of `bs-let`, I've actually never used it before but feel free to add it if it makes sense for the rewrite -- my impression is that when there are a lot of nested callbacks `bs-let` is useful, but I'm not sure if that will be the case here.
